src/components/remote-promo-sheet/checkForCampaign.ts

```diff
@@ -0,0 +1,157 @@
+  isAfterCampaignLaunch = 'isAfterCampaignLaunch',
+  hasNonZeroTotalBalance = 'hasNonZeroTotalBalance',
+}
+
+export type CampaignCheckResult = {
+  campaignId: string;
+  campaignKey: string;
+};
+
+export const checkForCampaign = async () => {
+  logger.log('Campaigns: Running Checks');
+
+  const { promoSheetCollection } = await fetchPromoSheetCollection({});
+
+  const campaignPromises = (promoSheetCollection?.items || [])
+    .filter((campaign): campaign is PromoSheet => campaign !== null)
+    .map(async campaign => await shouldPromptCampaign(campaign));
+
+  // In order to save computational bandwidth, we will resolve with the first campaign that should be shown, disregarding all others
+  const result = await resolveFirstRejectLast(campaignPromises);
+  if (!result) {
+    logger.info(`No campaign to prompt`);
+    return;
+  }
+
+  const isPromoCurrentlyShown = mmkv.getBoolean(
+    STORAGE_IDS.PROMO_CURRENTLY_SHOWN
+  );
+
+  // another sanity check for making sure we don't stack promo sheets
+  if (isPromoCurrentlyShown) return;
+  triggerCampaign(result);
+};
+
+export const triggerCampaign = async ({
+  campaignId,
+  campaignKey,
+}: CampaignCheckResult) => {
+  logger.log(`Campaign: Showing ${campaignKey} Promo`);
+  mmkv.set(campaignKey, true);
+  mmkv.set(STORAGE_IDS.PROMO_CURRENTLY_SHOWN, true);
+  mmkv.set(STORAGE_IDS.LAST_PROMO_SHEET_TIMESTAMP, Date.now());
+
+  setTimeout(() => {
+    InteractionManager.runAfterInteractions(() => {
+      Navigation.handleAction(Routes.REMOTE_PROMO_SHEET, {
+        campaignId,
+        campaignKey,
+      });
+    });
+  }, 1000);
+};
+
+export const shouldPromptCampaign = async (
+  campaign: PromoSheet
+): Promise<CampaignCheckResult | undefined> => {
+  const {
+    campaignKey,
+    sys: { id },
+    actions,
+  } = campaign;
+
+  // if we aren't given proper campaign data or actions to check against, exit early here
+  if (!campaignKey || !id || !actions.length) return;
+
+  // sanity check to prevent showing a campaign twice to a user or potentially showing a campaign to a fresh user
+  const hasViewedCampaign = mmkv.getBoolean(campaignKey as string);
+  const firstLaunch = fns.isFirstLaunch(mmkv);
+
+  logger.log(`Checking if we should prompt campaign ${campaignKey}`);
+  logger.info(`Has Viewed Campaign: ${hasViewedCampaign}`);
+  logger.info(`Is First Launch: ${firstLaunch}`);
+
+  // If the campaign has been viewed already or it's the first app launch, exit early
+  if (hasViewedCampaign || firstLaunch) {
+    return;
+  }
+
+  const shouldPrompt = (
+    await Promise.all(
+      ((actions || []) as ActionObj[]).map(
+        async ({ fn, outcome, props = {} }) => {
+          const action = __INTERNAL_ACTION_CHECKS[fn];
+          logger.info(
+            `Checking action: ${fn}. Typeof action: ${typeof action}`
+          );
+          if (typeof action === 'undefined') return false;
+
+          const params = (await __INTERNAL_ACTION_PROPS[fn]?.(
```

Comment:
honestly just kinda hard to read worth adding some comments here I think

Reply:
these are just for internal localized props per check-fn. I'm probably going to end up removing this since prop fetching is technically able to be done in each function itself.